src/actions/actions.ts

```diff
@@ -1960,6 +1960,19 @@ class ActionDeleteChar extends BaseCommand {
 }
 
 @RegisterAction
+class ActionDeleteCharWithDeleteKey extends ActionDeleteChar {
+  keys = ["<delete>"];
+
+  public async execCount(position: Position, vimState: VimState): Promise<VimState> {
+    if (vimState.recordedState.count !== 0) {
```

Comment:
This is actually impossible, as 0 by itself is a motion, so it can't prefix any motion.